datafusion/row/src/lib.rs

```diff
@@ -221,6 +221,56 @@ mod tests {
         vec![Some(5), Some(7), None, Some(0), Some(111)]
     );
 
+    #[test]
+    #[allow(non_snake_case)]
+    fn test_single_decimal128() -> Result<()> {
+        let v = vec![
+            Some(0),
+            Some(1),
+            None,
+            Some(-1),
+            Some(i128::MIN),
+            Some(i128::MAX),
+        ];
+        let schema =
+            Arc::new(Schema::new(vec![Field::new("a", Decimal128(38, 10), true)]));
+        let a = Decimal128Array::from(v);
+        let batch = RecordBatch::try_new(schema.clone(), vec![Arc::new(a)])?;
+        let mut vector = vec![0; 1024];
```

Comment:
Is there some way to ensure that we have enough space rather than using 1024 bytes? I am imagining some future bug or something that accidentally increases the write size for a decimal that will cause this to overflow and thus cause the tests to segfault rather than a nicer error message.